internal/testrunner/testrunner.go

```diff
@@ -26,10 +30,28 @@ type TestOptions struct {
 }
 
 // RunFunc method defines main run function of a test runner.
-type RunFunc func(options TestOptions) error
+type RunFunc func(options TestOptions) ([]TestResult, error)
 
 var runners = map[TestType]RunFunc{}
 
+// TestResult contains a single test's results
+type TestResult struct {
+	Name string
+
+	Package    string
+	DataStream string
+
+	TimeTaken time.Duration
+
+	FailureMsg string
+	ErrorMsg   string
```

Comment:
I suppose these two fields need a clarification.

Reply:
Added explanation comments in 27c42f9.

Reply:
Hm.. I think somebody may have a follow up question: what is a failure and what is an error? What is the difference?

Reply:
Okay, I will try to clarify this further in the comment. Thanks.

Reply:
Done in 848a72f132f90a2ecf21c4c6a6af8176e4400537.